pgt/src/PGT/Output/Text.hs

```diff
@@ -0,0 +1,35 @@
+module PGT.Output.Text
+  ( parseName
+  )
+where
+
+import Data.Attoparsec.Text (Parser)
+import PGT.Prelude
+
+import qualified Data.Attoparsec.Text as Text
+import qualified Data.Char            as Char
+import qualified Data.Text            as Text
+
+parseName :: Parser Text
+parseName = do
+  head <- Text.takeWhile1 Char.isAlphaNum
+  tail <- Text.many' parseTail
+
+  pure . Text.concat $ head : tail
+  where
+    parseTail :: Parser Text
+    parseTail = do
+      char      <- parseSeparator
+      remaining <- Text.takeWhile1 Char.isAlphaNum
+
+      pure $ Text.singleton char <> remaining
+      where
+        parseSeparator :: Parser Char
+        parseSeparator
+          = Text.choice
+          $ Text.char <$>
+          [ '-'
+          , '.'
+          , '_'
+          , ' '
+          ]
```

Comment:
__question:__ is the intention with this function to parse one or more alpha numeric characters, then one separator, and then one or more alpha numeric characters? Does the separator only appear one time in the name?